src/components/checkboxFilter/index.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+
+import { chakra } from '@chakra-ui/react';
+
+import Stack from '../stack';
+import Checkbox from '../checkbox';
+
+type Item<Option> = {
+  value: Option;
+  label: string;
+};
+
+type Props<Option extends string | number> = {
+  /**
+   * Name of the filter - only used on the DOM.
+   */
+  name: string;
+  /**
+   * @param Item: { value: Option, label: string }
+   */
+  options: Item<Option>[];
+  /**
+   * The checkbox filter is a controlled component and the updated filter value must be passed in order to see the correct state.
+   */
+  checked: { [key in Option]: boolean };
+  /**
+   * Numeric value shown next to the checkbox label. Indicates how many search results are going to be visible after the checkbox has been applied.
+   */
+  facets: { [key in Option]: number };
+  /**
+   * Callback function that is triggered after any checkbox has been clicked.
+   * @param item: contains the modified checkbox with the new value
+   * @param newFilterState: contains the new state of the whole filter group
+   */
+  onApply: (
+    item: { value: Option; isChecked: boolean },
+    newFilterState: { [key in Option]: boolean }
+  ) => void;
+};
+
+const addThousandSeparatorToNumber = (value: number) => {
+  return value.toString().replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1’');
+};
+
+function CheckboxFilter<Option extends string | number>({
```

Comment:
I see there is ChakraUI `CheckboxGroup` component and `useCheckboxGroup` custom hook. Is it worth and possible to use it for our case?

Reply:
I checked this but I think this achieves a different purpose. The hook should be used to use your own HTML and not Chakra components so that you can easily apply the props. The component is more for this usecase with the nesting: https://main-components-pkg.branch.autoscout24.dev/?path=/story/components-forms-checkbox-group--overview

but maybe I got it wrong?